modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/SortedIndexStorage.java

```diff
@@ -58,12 +64,63 @@
      */
     void remove(IndexRow row);
 
+    /** Exclude lower bound. */
+    byte GREATER = 0;
+
+    /** Include lower bound. */
+    byte GREATER_OR_EQUAL = 1;
+
+    /** Exclude upper bound. */
+    byte LESS = 0;
+
+    /** Include upper bound. */
+    byte LESS_OR_EQUAL = 1 << 1;
+
+    byte FORWARD = 0;
+
+    byte BACKWARDS = 1 << 2;
```

Comment:
Does you mean that every index storage MUST support Backward iteration?
If it is not mandatory requirement, then let's add smth like UnsupportedBackwardIndexScanException to method contract into "@throws" section.